### PR TITLE
rose edit: added option to suppress default version to HEAD warnings

### LINKED ISSUE #1904: rose edit: no warnings for default-version-to-HEAD
For some `rose-stem` suites, each app may be at a version which does not have its own metadata (and will therefore default to HEAD). Currently, a warning dialog will pop up for each app that does this. This can be a bit annoying for a large `rose-stem` suite.

It would be good to have a `--no-warn-version` option (or similar) to disable the dialog.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1904 

@benfitzpatrick Please Review
@matthewrmshin Please Review

### CODE REVIEW

File: lib/python/rose/config_editor/main.py [hunk 1]
```diff
@@ -1872,7 +1872,8 @@ def spawn_window(config_directory_path=None, debug_mode=False,
                               load_all_apps=load_all_apps,
                               load_no_apps=load_no_apps,
                               metadata_off=metadata_off,
-                              opt_meta_paths=opt_meta_paths)
+                              opt_meta_paths=opt_meta_paths,
+                              no_warn_version=no_warn_version)
```
Comment (oliver-sanders) [inline lib/python/rose/config_editor/main.py]: Perhaps pass opts to MainController to cut down on kwargs?

File: lib/python/rose/macro.py [hunk 2]
```diff
@@ -48,7 +48,10 @@
 ERROR_LOAD_CONFIG_DIR = "{0}: not an application or suite directory.\n"
 ERROR_LOAD_MACRO = "Could not load macro {0}: {1}"
 ERROR_LOAD_METADATA = "Could not load metadata {0}\n"
-ERROR_LOAD_CHOSEN_META_PATH = "Could not find metadata for {0}, using {1}\n"
+ERROR_LOAD_CHOSEN_META_PATH = (
+    "Could not find metadata for {0}, using {1}\n"
+    "To surpress these warnings run 'rose edit %s'\n" % (
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Typo

File: lib/python/rose/opt_parse.py [hunk 3]
```diff
@@ -336,6 +336,12 @@ class RoseOptionParser(OptionParser):
             {"action": "store_true",
              "dest": "no_headers",
              "help": "Suppress headers."}],
+        "no_warn_version": [
+            ["--no-warn-version"],
+            {"action": "store_true",
+             "default": False,
+             "help": "Surpress \"could not find metadata for app/version, " +
```
Comment (benfitzpatrick) [inline lib/python/rose/opt_parse.py]: Suppress typo

**Comment by benfitzpatrick** [comment]
Looks good, one typo

**Comment by oliver-sanders** [comment]
@benfitzpatrick Typo corrected.

**Comment by matthewrmshin** [comment]
Looks OK.

(On the new `--no-warn-version` option, I am wondering if it it more desirable to have a more flexible option for future extension. Examples can be found in warning options in compilers, e.g. `-W[no-]keyword` in `gcc`, `-warn keyword` and `-nowarn keyword` in `ifort`, etc.)

**Comment by oliver-sanders** [comment]
@matthewrmshin Changed usage from `--no-warn-version` to `--no-warn version`

File: bin/rose-config-edit [hunk 4]
```diff
@@ -50,16 +50,21 @@
 #        Launch, ignoring any configuration.
 #     --no-metadata
 #        Launch with metadata switched off.
-#     --no-warn-version
-#        Suppress "could not find metadata for app/version, using app/HEAD"
-#        warnings.
+#     --no-warn
```
Comment (matthewrmshin) [inline bin/rose-config-edit]: Should say `--no-warn=WARNING-TYPE` so it is clear to user that the option expects an argument.

File: lib/python/rose/opt_parse.py [hunk 5]
```diff
@@ -665,7 +665,12 @@ class RoseOptionParser(OptionParser):
             {"action": "store_true",
              "dest": "view_mode",
              "default": False,
-             "help": "View with web browser."}]}
+             "help": "View with web browser."}],
+        "no_warn": [
+            ["--no-warn"],
+            {"action": "append",
+             "dest": "no_warn",
+             "help": "Warnings to disable."}]}
```
Comment (matthewrmshin) [inline lib/python/rose/opt_parse.py]: I think you can add `"choice": ["version"]` here to restrict the allowed keyword.

You should also an a `"meta": "WARNING-TYPE"` here to indicate that the option has arguments.

**Comment by oliver-sanders** [comment]
@matthewrmshin Addressed comments.

**Comment by matthewrmshin** [comment]
Looks good.

@benfitzpatrick please sanity check the latest user interface, as it has been modified since you last looked at it.

File: lib/python/rose/macro.py [hunk 6]
```diff
@@ -451,6 +453,8 @@ def load_meta_path(config=None, directory=None, is_upgrade=False,
     """
     if config is None:
         config = rose.config.ConfigNode()
+    if no_warn is None:
+            no_warn = []
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Over-indented

Comment (oliver-sanders) [inline lib/python/rose/macro.py]: @benfitzpatrick Resolved

File: lib/python/rose/macro.py [hunk 7]
```diff
@@ -48,7 +48,9 @@
 ERROR_LOAD_CONFIG_DIR = "{0}: not an application or suite directory.\n"
 ERROR_LOAD_MACRO = "Could not load macro {0}: {1}"
 ERROR_LOAD_METADATA = "Could not load metadata {0}\n"
-ERROR_LOAD_CHOSEN_META_PATH = "Could not find metadata for {0}, using {1}\n"
+ERROR_LOAD_CHOSEN_META_PATH = (
+    "Could not find metadata for {0}, using {1}\n"
+    "To suppress these warnings run 'rose edit --no-warn version")
```
Comment (benfitzpatrick) [inline lib/python/rose/macro.py]: Sorry, just ran a final walk through and noticed the missing `'` after `version`.

**Comment by benfitzpatrick** [comment]
We also get the `rose edit --no-warn=version` advice from `rose macro`

**Comment by oliver-sanders** [comment]
Add this functionality to rose macro?

**Comment by benfitzpatrick** [comment]
Actually, why not - yes

**Comment by benfitzpatrick** [comment]
#1891 has caused a conflict, I think.

**Comment by oliver-sanders** [comment]
@benfitzpatrick PR de-conflicted.

**Comment by oliver-sanders** [comment]
@matthewrmshin 91fe45c93d5d827881f348690ead953d626896af fixes rose documentation test issues.

**Pull request merged**